### PR TITLE
Selectors for heatmaps that are too big.

### PR DESCRIPTION
**Pull request by gregory-marton**

### CODE REVIEW

File: src/plot_utils.py [hunk 1]
```diff
@@ -96,7 +102,18 @@ def heatmap(bdb, bql, **kwargs):
...
+    if selectors is None:
+        return zmatrix(df, **kwargs)
+    results = []
+    for n0selector in selectors:
+      n0selection = deps['name0'].map(n0selector)
+      for n1selector in selectors:
+        n1selection = deps['name1'].map(n1selector)
+        this_block = deps[n0selection & n1selection]
+        results.append(zmatrix(this_block, vmin=0, vmax=1))
+        plt.show()
```
Comment (axch) [inline src/plot_utils.py]: How will this play out in interactive but not IPython use?
Will it be possible to draw those plots headless and unattended and save them to files?

Comment (riastradh-probcomp) [inline src/plot_utils.py]: Date: Tue, 29 Sep 2015 18:14:05 -0700
   From: axch <notifications@github.com>

   > @@ -96,7 +102,18 @@ def heatmap(bdb, bql, **kwargs):
   >      """
   >      df = bqlu.cursor_to_df(bdb.execute(bql))
   >      df.fillna(0, inplace=True)
   > -    return zmatrix(df, **kwargs)
   > +    if selectors is None:
   > +        return zmatrix(df, **kwargs)
   > +    results = []
   > +    for n0selector in selectors:
   > +      n0selection = deps['name0'].map(n0selector)
   > +      for n1selector in selectors:
   > +        n1selection = deps['name1'].map(n1selector)
   > +        this_block = deps[n0selection & n1selection]
   > +        results.append(zmatrix(this_block, vmin=0, vmax=1))
   > +        plt.show()

   How will this play out in interactive but not IPython use?
   Will it be possible to draw those plots headless and unattended and save them to files?

I'm unclear on the context, but all bdbcontrib plotting utilities
should be factored into one generic back end routine to create a
figure object, and another front-end-specific routine (e.g., in the
shell) to actually draw it as part of a user interface.  This appears
to violate that separation.

Comment (fsaad) [inline src/plot_utils.py]: I agree -- I missed the `plt.show` in code review a

File: src/plot_utils.py [hunk 2]
```diff
@@ -96,7 +102,18 @@ def heatmap(bdb, bql, **kwargs):
...
+        plt.close('all')
```
Comment (fsaad) [inline src/plot_utils.py]: Although here we close the plot?

Comment (gregory-marton) [inline src/plot_utils.py]: I agree with you both, and plan to remove it, just haven't gotten around to
it. Thanks for noticing!


Gregory A. Marton  617-858-0775
http://goo.gl/Ne09o
http://csail.mit.edu/~gremio

Comment (gregory-marton) [inline src/plot_utils.py]: Resolved in https://github.com/probcomp/bdbcontrib/pull/70


Gregory A. Marton  617-858-0775
http://goo.gl/Ne09o
http://csail.mit.edu/~gremio

**Pull request merged**